src/worker/abstract-worker.ts

```diff
@@ -134,7 +128,7 @@ export abstract class AbstractWorker<
    * Check to see if the worker should be terminated, because its living too long.
    */
   protected checkAlive (): void {
-    if (Date.now() - this.lastTask > this.maxInactiveTime && !this.isKilled) {
```

Comment:
isKilled was not intentionally pushed to the master branch, since is not fixing the problem I removed it.